Adds Advancing the Web Framework Ecosystem CDS article

housseindjirdeh opened the pull request:
Fixes #2017.

Changes proposed in this pull request:

- Includes article for the "Advancing the web framework ecosystem" talk at CDS

(Note: First review already completed in Google Doc by @mihajlija)

CC @spanicker

kaycebasques commented on src/site/content/en/blog/advancing-framework-ecosystem-cds-2019/index.md:
```diff
@@ -0,0 +1,208 @@
+date: 2020-01-08
+description: |
+  Learn about how Chrome is investing efforts in a number of open-source tools to advance the JavaScript ecosystem
+hero: hero.jpg 
```
wow much fancy v pleasant

kaycebasques commented on src/site/content/en/blog/advancing-framework-ecosystem-cds-2019/index.md:
```diff
@@ -0,0 +1,208 @@
+    [MDN's First Annual Web Developer & Designer Survey](https://hacks.mozilla.org/2019/12/presenting-the-mdn-web-developer-needs-assessment-web-dna-report/)
+    use at least one of these frameworks and libraries.
++   Over
+    [320,000 sites](https://bigquery.cloud.google.com/savedquery/1086077897885:24ffb259f2a04a7f9955e44f6e0298e9)in
```
```suggestion
    [320,000 sites](https://bigquery.cloud.google.com/savedquery/1086077897885:24ffb259f2a04a7f9955e44f6e0298e9) in
```

Adds space between `sites` and `in`

kaycebasques commented on src/site/content/en/blog/advancing-framework-ecosystem-cds-2019/index.md:
```diff
@@ -0,0 +1,208 @@
+</figure>
+
+Bundle size reduction for angular.io with differential builds
+(from[" Version 8 of Angular"](https://blog.angular.io/version-8-of-angular-smaller-bundles-cli-apis-and-alignment-with-the-ecosystem-af0261112a27))
```
Looks like this caption was accidentally duplicated so need to delete lines 85 & 86

![image](https://user-images.githubusercontent.com/4713486/72032926-a7f1df00-3245-11ea-9d8c-24dcf99d26d2.png)

kaycebasques commented on src/site/content/en/blog/advancing-framework-ecosystem-cds-2019/index.md:
```diff
@@ -0,0 +1,208 @@
+(from[" Version 8 of Angular"](https://blog.angular.io/version-8-of-angular-smaller-bundles-cli-apis-and-alignment-with-the-ecosystem-af0261112a27))
+
++   Support for standard dynamic import syntax for lazy-loading routes.
++   [Web worker support](https://angular.io/guide/web-worker).
```
Can you expand this sentence a little bit to explain why this is good

kaycebasques commented on src/site/content/en/blog/advancing-framework-ecosystem-cds-2019/index.md:
```diff
@@ -0,0 +1,208 @@
++   [Web worker support](https://angular.io/guide/web-worker).
++   [Ivy](https://www.youtube.com/watch?v=jnp_ny4SOQE&feature=youtu.be&t=1320), Angular's new
+    rendering engine, is available in [preview
+    mode](https://angular.io/guide/ivy#opting-into-angular-ivy) for existing projects.
```
Can you also expand this sentence to explain why Ivy is an improvement

kaycebasques commented on src/site/content/en/blog/advancing-framework-ecosystem-cds-2019/index.md:
```diff
@@ -0,0 +1,208 @@
+<figure class="w-figure w-figure--center">
+  <img src="barnebys.png" class="w-screenshot" alt="Homepage of Barnebys.com">
+  <figcaption class="w-figcaption">
+    <a href="https://www.barnebys.com/">Barnebys.com</a>, a large search engine for antiques and collectibles, saw a 23% reduction in total JavaScript with granular chunking enabled
```
```suggestion
    <a href="https://www.barnebys.com/">Barnebys.com</a>, a large search engine for antiques and collectibles, saw a 23% reduction in total JavaScript after enabling granular chunking
```

Rephrasing this sentence slightly to emphasize the action that created the improvement (this is kind of a nit but please just trust my writer intuition on this)

kaycebasques commented on src/site/content/en/blog/advancing-framework-ecosystem-cds-2019/index.md:
```diff
@@ -0,0 +1,208 @@
+    surface better errors and warnings ([RFC](https://github.com/zeit/next.js/issues/9310)).
+
+<figure class="w-figure w-figure--center">
+  <img src="conformance.png" class="w-screenshot" alt="Example of a conformance build error in Next.js">
```
```suggestion
  <img src="conformance.png" class="w-screenshot-filled" alt="Example of a conformance build error in Next.js">
```

This will give it a nice grey background

![image](https://user-images.githubusercontent.com/4713486/72033227-89d8ae80-3246-11ea-8676-7da802eee2b5.png)

Compared to the somewhat funky grey border currently:

![image](https://user-images.githubusercontent.com/4713486/72033244-9c52e800-3246-11ea-8ed3-255552d325ca.png)

kaycebasques commented on src/site/content/en/blog/advancing-framework-ecosystem-cds-2019/index.md:
```diff
@@ -0,0 +1,208 @@
++   Default data store ([Vuex](https://vuex.vuejs.org/guide/))
+
+Along with working directly on improving the performance of different tools, we've expanded the
+framework fund to provide monetary support to more open-source frameworks and libraries. With our
```
```suggestion
[framework fund](https://opencollective.com/chrome) to provide monetary support to more open-source frameworks and libraries. With our
```

Ignore this if I'm wrong, but this seems to be the first mention of `framework fund`, so this is where we should link to more info

kaycebasques commented on src/site/content/en/blog/advancing-framework-ecosystem-cds-2019/index.md:
```diff
@@ -0,0 +1,208 @@
+optimizations.
+
+{% Aside %}
+The [framework fund](https://opencollective.com/chrome) accelerates the efforts of different
```
```suggestion
The framework fund accelerates the efforts of different
```

Link can be removed since we moved it up in the previous paragraph

kaycebasques commented on src/site/content/en/blog/advancing-framework-ecosystem-cds-2019/index.md:
```diff
@@ -0,0 +1,208 @@
+{% Aside %}
+The [framework fund](https://opencollective.com/chrome) accelerates the efforts of different
+frameworks and libraries with the goal to improve their performance. If you are working on an
+open-source tool and need support, [please apply](http://bit.ly/chrome-framework-fund)!
```
```suggestion
open-source tool and need support, [please apply](https://bit.ly/chrome-framework-fund)!
```

Changing link to HTTPS

Nice call-to-action tho

kaycebasques commented on src/site/content/en/blog/advancing-framework-ecosystem-cds-2019/index.md:
```diff
@@ -0,0 +1,208 @@
+
+{% Aside %}
+To learn more about how to leverage ES modules to minimize polyfill code shipped to browsers
+that do not need them check out
```
```suggestion
that do not need them, check out
```

kaycebasques commented on src/site/content/en/blog/advancing-framework-ecosystem-cds-2019/index.md:
```diff
@@ -0,0 +1,208 @@
+{% Aside %}
+To learn more about how to leverage ES modules to minimize polyfill code shipped to browsers
+that do not need them check out
+["Serve modern code to modern browsers for faster page loads"](../serve-modern-code-to-modern-browsers).
```
```suggestion
[Serve modern code to modern browsers for faster page loads](/serve-modern-code-to-modern-browsers).
```

No need for quotes around titles because the link styling sets them apart

Also changed the path to be absolute

kaycebasques commented on src/site/content/en/blog/advancing-framework-ecosystem-cds-2019/index.md:
```diff
@@ -0,0 +1,208 @@
+To optimize for this case, we launched a brand new preset;
+[@babel/preset-modules](http://github.com/babel/preset-modules). Instead of converting modern syntax
+to older syntax to avoid browser bugs, preset-modules fixes each specific bug by transforming to the
+closest possible non-broken modern syntax. This results in modern code that can be delivered_ nearly
```
```suggestion
closest possible non-broken modern syntax. This results in modern code that can be delivered *nearly
```

italics isn't working

![image](https://user-images.githubusercontent.com/4713486/72033498-58acae00-3247-11ea-8935-bd5544099545.png)

kaycebasques commented on src/site/content/en/blog/advancing-framework-ecosystem-cds-2019/index.md:
```diff
@@ -0,0 +1,208 @@
+[@babel/preset-modules](http://github.com/babel/preset-modules). Instead of converting modern syntax
+to older syntax to avoid browser bugs, preset-modules fixes each specific bug by transforming to the
+closest possible non-broken modern syntax. This results in modern code that can be delivered_ nearly
+unmodified _to most browsers.
```
```suggestion
unmodified* to most browsers.
```

kaycebasques commented on src/site/content/en/blog/advancing-framework-ecosystem-cds-2019/index.md:
```diff
@@ -0,0 +1,208 @@
+
+To optimize for this case, we launched a brand new preset;
+[@babel/preset-modules](http://github.com/babel/preset-modules). Instead of converting modern syntax
+to older syntax to avoid browser bugs, preset-modules fixes each specific bug by transforming to the
```
```suggestion
to older syntax to avoid browser bugs, `preset-modules` fixes each specific bug by transforming to the
```

We usually style tool names as inline code, it's just a minor style thing to keep the post consistent with the rest of the site

kaycebasques commented on src/site/content/en/blog/advancing-framework-ecosystem-cds-2019/index.md:
```diff
@@ -0,0 +1,208 @@
+
+![A new babel preset to provide better polyfilling for browsers](./babel-preset-env.png)
+
+Developers who already use preset-env will also benefit from these optimizations without having to
```
```suggestion
Developers who already use `preset-env` will also benefit from these optimizations without having to
```

Styling tool name as code, same reason as other fix above

kaycebasques commented on src/site/content/en/blog/advancing-framework-ecosystem-cds-2019/index.md:
```diff
@@ -0,0 +1,208 @@
+Chrome team realize what is fundamentally important to users and developers alike.
+
+If you work on a web framework, UI library, or any form of web tooling (bundler, compiler, linter),
+[reach out to us](https://twitter.com/ChromiumDev)!
```
```suggestion
[apply for the framework fund](http://bit.ly/chrome-framework-fund)!
```
Maybe a more powerful call-to-action? Feel free to ignore.

housseindjirdeh replied:
Agreed 👍

kaycebasques commented on src/site/content/en/blog/advancing-framework-ecosystem-cds-2019/index.md:
```diff
@@ -0,0 +1,208 @@
+<figure class="w-figure w-figure--center">
+  <img src="differential-loading-angular.png" class="w-screenshot" alt="Graph showing bundle size reduction of angular.io with and without differential builds">
+  <figcaption class="w-figcaption">
+    Bundle size reduction for angular.io with differential builds (from <a href="https://blog.angular.io/version-8-of-angular-smaller-bundles-cli-apis-and-alignment-with-the-ecosystem-af0261112a27">"Version 8 of Angular"</a>)
```
```suggestion
    Bundle size reduction for angular.io with differential builds (from <a href="https://blog.angular.io/version-8-of-angular-smaller-bundles-cli-apis-and-alignment-with-the-ecosystem-af0261112a27">Version 8 of Angular</a>)
```

Remove quotes from title, because links are styled blue that sufficiently sets them apart and there's no need for quotes

housseindjirdeh commented:
@kaycebasques Thanks a million for reviewing! Fixed all the nits and updated per your suggestions.

> Please try to do thorough reviews of Markdown, e.g. there was a duplicated caption, and some italic text that didn't format properly, and a missing space between two words

My apologies! Using a docs to markdown converter always has a few of these nits, I unfortunately missed fixing these few.

> Please push to a branch on googlechrome/web.dev

Will do from now on!

kaycebasques commented:
@housseindjirdeh FYI in case you're not aware when you see these **Commit Suggestion** UIs it means that I have provided the fix to the code and if you click that **Commit Suggestion** button you can update the branch through the GitHub UI. It looks like you manually fixed each change the old-fashioned way. You probably already know about this UI I'm just making sure that you're aware of this potentially faster workflow.

![image](https://user-images.githubusercontent.com/4713486/72188493-54030980-33af-11ea-88bd-55e9c849b590.png)
